pkg/k8s/function_status.go

```diff
@@ -48,8 +51,32 @@ func AsFunctionStatus(item appsv1.Deployment) *types.FunctionStatus {
 	for _, v := range functionContainer.Env {
 		if EnvProcessName == v.Name {
 			function.EnvProcess = v.Value
+			continue
 		}
+		function.EnvVars[v.Name] = v.Value
 	}
 
 	return &function
 }
+
+func nodeSelectorToConstraint(deploy appsv1.Deployment) []string {
+	nodeSelector := deploy.Spec.Template.Spec.NodeSelector
```

Comment:
Could NodeSelector be nil?

Reply:
NodeSelector is a `map[string]string`, so it _can_ be `nil`, however, we `range` over it, which is safe, the `range` becomes a no-op.

You can see that it is safe from this https://play.golang.com/p/s4CyxcqElaA

```go
package main

import (
	"fmt"
)

func main() {
	var info map[string]string
	for key, value := range info {
		fmt.Printf("%s=%s\n", key, value)
	}
	fmt.Println("Hello, playground")
}
```

Reply:
Thanks